apis/placement/v1beta1/clusterresourceplacement_types.go

```diff
@@ -429,14 +429,116 @@ type RolloutStrategy struct {
 // Note: If multiple CRPs try to place the same resource with different apply strategy, the later ones will fail with the
 // reason ApplyConflictBetweenPlacements.
 type ApplyStrategy struct {
-	// Type defines the type of strategy to use. Default to ClientSideApply.
-	// Server-side apply is more powerful and flexible than client-side apply.
-	// You SHOULD use server-side apply to safely resolve any potential drift between the
-	// original applied resource version and the current resource on the member cluster.
-	// Read more about the differences between server-side apply and client-side apply:
-	// https://kubernetes.io/docs/reference/using-api/server-side-apply/#comparison-with-client-side-apply.
-	// You can also use ReportDiff to only report the difference between the resource on the member cluster
-	// and the resource to be applied from the hub on all the selected clusters.
+	// CompareOption controls how Fleet compares the desired state of a resource, as kept in
+	// a hub cluster manifest, with the current state of the resource (if applicable) in the
+	// member cluster.
+	//
+	// Available options are:
+	//
+	// * PartialDiff: with this option, Fleet will compare only fields that are managed by Fleet, i.e.,
+	//   the fields that are specified explicitly in the hub cluster manifest. Unmanaged fields
+	//   are ignored. This is the default option.
+	//
+	// * FullDiff: with this option, Fleet will compare all fields of the resource, even if the
+	//   fields are absent from the hub cluster manifest.
+	//
+	// Consider using the PartialDiff option if you would like to:
+	//
+	// * use the default values for certain fields; or
+	// * let another agent, e.g., HPAs, VPAs, etc., on the member cluster side manage some fields; or
+	// * allow ad-hoc or cluster-specific settings on the member cluster side.
+	//
+	// To use the FullDiff option, it is recommended that you:
+	//
+	// * specify all fields as appropriate in the hub cluster, even if you are OK with using default
+	//   values;
+	// * make sure that no fields are managed by agents other than Fleet on the member cluster
+	//   side, such as HPAs, VPAs, or other controllers.
+	//
+	// See the Fleet documentation for further explanations and usage examples.
+	//
+	// +kubebuilder:default=PartialDiff
+	// +kubebuilder:validation:Enum=PartialDiff;FullDiff
+	// +kubebuilder:validation:Optional
+	CompareOption CompareOptionType `json:"compareOption,omitempty"`
```

Comment:
Use ComparisonOption: FullComparison, PartialComparison.

Or use DiffOption: PartialDiff, FullDiff.

Reply:
I think ComparisonOption is better as it will apply to both diff and drift

Reply:
Done 👌